lib/scripts/mtw.js

```diff
@@ -292,6 +301,36 @@ export class ContentScript {
   }
 
   /**
+   * Translate one word in each sentence for a  paragraph.
+   * @param {Object} paragraph - Paragraph nodeType
+   * @param {Object} filteredTMap - filtered translation map
+   * @param {Object} iMap - HTML element for each translated word
+   */
+  translateOneWord(paragraph, filteredTMap, iMap) {
+    for (let i in paragraph.childNodes) {
+      if (paragraph.childNodes[i].nodeType === 3) {
+        if(!/^\s*$/.test(paragraph.childNodes[i].textContent)){
+          let sentences = paragraph.childNodes[i].textContent.split('.');
+          for (let j in sentences) {
+            let words = sentences[j].split(' ');
+            words = _.shuffle(words);
+            for (let k in words) {
+              if (filteredTMap[words[k]]) {
+                let x = sentences[j].replace(words[k], iMap[filteredTMap[words[k]]]);
+                sentences[j] = x;
+                break;
+              }
+            }
+          }
+          var newNode = document.createElement('span');
+          newNode.innerHTML = sentences.join('.');
```

Comment:
Ah, ok! I (wrongly) thought that the spaces would be removed by the other split, in line 315. But now I see that, in lines 319 and 320, you do not construct the new sentence by joining `words`, and therefore the removal of spaces in `words` won't matter.

By the way, this made me wonder: maybe reconstructing the sentence by joining `words` would be slightly more efficient? Right now, the "replace" in line 319 is traversing the whole sentence, even though we already know that it is the k-th word that has to be replaced.